alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultAuthorizationUtility.java

```diff
@@ -0,0 +1,137 @@
+import com.synopsys.integration.alert.database.authorization.PermissionMatrixRelation;
+import com.synopsys.integration.alert.database.authorization.PermissionMatrixRepository;
+import com.synopsys.integration.alert.database.user.RoleEntity;
+import com.synopsys.integration.alert.database.user.RoleRepository;
+import com.synopsys.integration.alert.database.user.UserRoleRelation;
+import com.synopsys.integration.alert.database.user.UserRoleRepository;
+
+@Component
+@Transactional
+public class DefaultAuthorizationUtility implements AuthorizationUtil {
+    private final RoleRepository roleRepository;
+    private final UserRoleRepository userRoleRepository;
+    private final PermissionMatrixRepository permissionMatrixRepository;
+    private final AccessOperationRepository accessOperationRepository;
+    private final PermissionKeyRepository permissionKeyRepository;
+
+    @Autowired
+    public DefaultAuthorizationUtility(final RoleRepository roleRepository, final UserRoleRepository userRoleRepository, final PermissionMatrixRepository permissionMatrixRepository, final AccessOperationRepository accessOperationRepository,
+        final PermissionKeyRepository permissionKeyRepository) {
+        this.roleRepository = roleRepository;
+        this.userRoleRepository = userRoleRepository;
+        this.permissionMatrixRepository = permissionMatrixRepository;
+        this.accessOperationRepository = accessOperationRepository;
+        this.permissionKeyRepository = permissionKeyRepository;
+    }
+
+    @Override
+    public Set<UserRoleModel> createRoleModels(final Collection<Long> roleIds) {
+        final Set<UserRoleModel> userRoles = new LinkedHashSet<>();
+        for (final Long roleId : roleIds) {
+            final Optional<String> roleName = getRoleName(roleId);
+            roleName.ifPresent(role -> userRoles.add(UserRoleModel.of(role, readPermissionsForRole(roleId))));
+        }
+        return userRoles;
+    }
+
+    @Override
+    public void updateUserRoles(final Long userId, final Collection<UserRoleModel> roles) {
+        if (null != userId) {
+            userRoleRepository.deleteAllByUserId(userId);
+        }
+
+        if (null != roles) {
+            final Collection<String> roleNames = roles.stream().map(UserRoleModel::getName).collect(Collectors.toSet());
+
+            final List<RoleEntity> roleEntities = roleRepository.findRoleEntitiesByRoleName(roleNames);
+            final List<UserRoleRelation> roleRelations = new LinkedList<>();
+            for (final RoleEntity role : roleEntities) {
+                roleRelations.add(new UserRoleRelation(userId, role.getId()));
+            }
+            userRoleRepository.saveAll(roleRelations);
+        }
+    }
+
+    private Optional<String> getRoleName(final Long roleId) {
+        return roleRepository.findById(roleId).map(RoleEntity::getRoleName);
+    }
+
+    @Override
+    public PermissionMatrixModel mergePermissionsForRoles(final Collection<String> roleNames) {
```

Comment:
I dont understand this method name. What is it merging?

Reply:
it merges the permissions for all the roles specified into one PermissionMatrixModel.  It's a superset of the permissions.